app/uk/gov/hmrc/nisp/controllers/pertax/PertaxHelper.scala

```diff
@@ -26,32 +27,46 @@ import uk.gov.hmrc.http.HeaderCarrier
 
 import scala.util.{Failure, Success}
 
-class PertaxHelper @Inject() (sessionCache: SessionCache, metricsService: MetricsService)(implicit
-  ec: ExecutionContext
+class PertaxHelper @Inject()(
+  sessionCache: SessionCache,
+  metricsService: MetricsService
+)(
+  implicit ec: ExecutionContext
 ) {
 
   def setFromPertax(implicit hc: HeaderCarrier): Unit = {
-    val timerContext = metricsService.keystoreWriteTimer.time()
-    val cacheF       = sessionCache.cache(PERTAX, true)
+    val keystoreTimerContext: Timer.Context =
+      metricsService.keystoreWriteTimer.time()
+    val cacheF: Future[CacheMap] =
+      sessionCache.cache(PERTAX, true)
+
     cacheF.onComplete {
-      case Success(_) => timerContext.stop()
-      case Failure(_) => metricsService.keystoreWriteFailed.inc()
+      case Success(_) =>
+        keystoreTimerContext.stop()
+      case Failure(_) =>
+        metricsService.keystoreWriteFailed.inc()
     }
   }
 
   def isFromPertax(implicit hc: HeaderCarrier): Future[Boolean] = {
-    val keystoreTimerContext = metricsService.keystoreReadTimer.time()
-    sessionCache.fetchAndGetEntry[Boolean](PERTAX).map { keystoreResult =>
-      keystoreTimerContext.stop()
-      keystoreResult match {
-        case Some(isPertax) => metricsService.keystoreHitCounter.inc(); isPertax
-        case None           =>
-          metricsService.keystoreMissCounter.inc()
-          false
-      }
-    } recover { case _ =>
-      metricsService.keystoreReadFailed.inc()
-      false
+    val keystoreTimerContext =
+      metricsService.keystoreReadTimer.time()
+
+    sessionCache.fetchAndGetEntry[Boolean](PERTAX).map {
+      keystoreResult =>
+        keystoreTimerContext.stop()
+        keystoreResult match {
+          case Some(isPertax) =>
+            metricsService.keystoreHitCounter.inc()
+            isPertax
+          case None =>
+            metricsService.keystoreMissCounter.inc()
+            false
+        }
+    } recover {
+      case _ =>
```

Comment:
Logging needed?

Reply:
This ticket is to remove the LocalTemplateRenderer and FormPartialRetriever and Play UI and other older UI assets, this knocked the coverage down, which meant I had to add tests, which meant I refactored this file to make it more readable. Functionality should be as was.